scripts/globusmonitor/globus_uploader_service.py

```diff
@@ -488,20 +383,20 @@ def notifyClowderOfCompletedTask(task):
                     if datasetMD:
                         md = {
                             "@context": ["https://clowder.ncsa.illinois.edu/contexts/metadata.jsonld",
-                                         {"@vocab": clowderContext}],
+                                         {"@vocab": clowder_context}],
                             "content": cleaned_dsmd,
                             "agent": {
                                 "@type": "cat:user",
-                                "user_id": "https://terraref.ncsa.illinois.edu/clowder/api/users/%s" % clowderId
+                                "user_id": "https://terraref.ncsa.illinois.edu/clowder/api/users/%s" % clowder_id
                             }
                         }
-                        dsmd = sess.post(clowderHost+"/api/datasets/"+dsid+"/metadata.jsonld",
+                        dsmd = sess.post(clowder_host+"/api/datasets/"+dsid+"/metadata.jsonld",
                                          headers={'Content-Type':'application/json'},
                                          data=json.dumps(md))
 
                         if dsmd.status_code != 200:
-                            logger.error("- cannot add dataset metadata (%s: %s)" % (dsmd.status_code, dsmd.text))
-                            return False
+                            logger.error("[%s] failed to attach metadata (%s: %s)" % (ds, dsmd.status_code, dsmd.text))
+                            response = "RETRY"
```

Comment:
Are you confident that all non-200 errors are retryable? Based on my experience, I'd limit to 502/504.